### PR TITLE
DOC: change branch and backport guidelines

### PR DESCRIPTION
**Pull request by tacaswell**
- Documents that we now use meeseeksdev
 - Documents current active branches
 - Documents 2.2.x backport guidelines

### CODE REVIEW

File: doc/devel/coding_guide.rst [hunk 1]
```diff
@@ -87,31 +84,67 @@ PR Review guidelines
...
+-------------------
 
-* ``DANGER`` is a read/write remote branch of the matplotlib/matplotlib repo
+We use meeseeksdev bot to automatically backport merges to the correct
+maintenance branch based on the milestone assigned to the PR.  The
+target branch is configured by putting ``on-merge: backport to
+TARGETBRANCH`` in the milestone description.
+
+The bot can also be manually triggered after a merge by leaving a
+message ``@meeseeksdev backport to BRANCH`` on the PR.
```
Comment (anntzer) [inline doc/devel/coding_guide.rst]: note that only whitelisted devs can use that (and mention how to be added?) (or did that change?)

Comment (QuLogic) [reply doc/devel/coding_guide.rst]: There is no longer a whitelist.

File: doc/devel/coding_guide.rst [hunk 2]
```diff
@@ -87,31 +84,67 @@ PR Review guidelines
...
+inconsistencies the user can work around in their code) are on a
+case-by-case basis, should be low-risk, and need someone to advocate
+for and shepherd through the backport.
+
+Automated backports
+-------------------
 
-* ``DANGER`` is a read/write remote branch of the matplotlib/matplotlib repo
+We use meeseeksdev bot to automatically backport merges to the correct
+maintenance branch based on the milestone assigned to the PR.  The
```
Comment (QuLogic) [inline doc/devel/coding_guide.rst]: Milestone should be assigned before merging.

File: doc/devel/coding_guide.rst [hunk 3]
```diff
@@ -87,31 +84,67 @@ PR Review guidelines
...
+======================
 
 
-When doing backports please include the branch you backported the
-commit to along with the SHA in a comment on the original PR.
+The current active branches are
 
-We do a backport from master to v2.0.x assuming:
+*master*
+  This will be Matplotlib 3.0.  Supports python 3.5+.
```
Comment (QuLogic) [inline doc/devel/coding_guide.rst]: Python

File: doc/devel/coding_guide.rst [hunk 4]
```diff
@@ -87,31 +84,67 @@ PR Review guidelines
...
 
-* ``matplotlib`` is a read-only remote branch of the matplotlib/matplotlib repo
+*v2.2.x*
+  Maintenance branch for Matplotlib 2.2 LTS.  Supports python 2.7, 3.4+
```
Comment (QuLogic) [inline doc/devel/coding_guide.rst]: Python

File: doc/devel/coding_guide.rst (hunk 1 above)
Comment (jkseppan) [reply doc/devel/coding_guide.rst]: Can you document what to do when meeseeksdev replies with the cryptic error message "Something went wrong ... Please have a look at my logs." Are the logs somewhere visible?

Comment (tacaswell) [reply doc/devel/coding_guide.rst]: I think Mathias has to add you to another service to have access to the logs.

File: doc/devel/coding_guide.rst [hunk 5]
```diff
@@ -87,31 +84,76 @@ PR Review guidelines
...
+  user can not work around)
+- fixes for regressions against 2.0 or 2.1
+
+Everything else (regressions against 1.x versions, bugs/api
+inconsistencies the user can work around in their code) are on a
+case-by-case basis, should be low-risk, and need someone to advocate
+for and shepherd through the backport.
+
+The only changes to be backported to 2.2.N-doc are changes to
+:path:`doc`, :path:`examples`, or :path:`totorials`.  Any changes to
```
Comment (QuLogic) [inline doc/devel/coding_guide.rst]: `tutorials`

**Comment by tacaswell** [comment]
self-merged on @dopplershift 's approval and discussion on the call today so it will be in the 2.2.2 docs which I am building now.

**Pull request merged**